lib/remap-lang/src/value.rs

```diff
@@ -0,0 +1,356 @@
+use bytes::Bytes;
+use std::collections::BTreeMap;
+use std::convert::{TryFrom, TryInto};
+use std::string::String as StdString;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Value {
+    String(Bytes),
```

Comment:
I think renaming this to `String` is confusing as it is still bytes.

I can see it does make for nicer error messages as we now get `expected type string` rather than `expected types bytes`, but I think we can still do that without changing the name here?

Reply:
Maybe. I like to be a bit more explicit about what type of value this is (it's similarly named `string` in the parser, not `bytes`).

It has to store `Bytes`, both for performance reasons, and because strings from an event might not be valid UTF-8.

Let's see how we like `String` for now. We can always do a refactor to rename to `Bytes` in a follow-up PR.